src/psyclone/tests/psyad/transformations/test_assignment_trans.py

```diff
@@ -49,6 +49,13 @@
 from psyclone.tests.utilities import Compile
 
 
+@pytest.fixture(name="index_str", params=["1", "i", ":", "2:3"])
```

Comment:
When I wrote the tests I made it so that the first two tests included different datatypes and then the remaining tests I left as using directly addressed arrays. In keeping with this I think it would be nice to add basic array notation examples to `test_zero()` and `test_single_assign()`. I certainly would not remove what you have done as it adds rigour, but I do have a question, which is why did you choose these particular tests? Is it just a randomly selected subset?

Reply:
I think it was just the tests where I thought it was appropriate. I've extended the tests as you suggest and added some more for edge cases where we have the same array but with different elements being accessed.